src/lib/y2network/connection_config/bonding.rb

```diff
@@ -34,6 +35,31 @@ def initialize
         @slaves = []
         @options = ""
       end
+
+      # @see Y2Network::InterfacesCollection
+      #
+      # @param interfaces [Y2Network::InterfacesCollection]
+      def update_interfaces!(interfaces)
+        return if slaves.empty?
+        new_slaves = []
+
+        slaves.each do |index, slave|
+          alternative = interfaces.by_name(slave)
```

Comment:
why alternative? do I get it correctly that it creates slaves if needed as fake interfaces if it is not found, right?

Reply:
It links the slaves with real or fake interfaces, yes. So definitely a bad naming.